trio/_core/_io_epoll.py

```diff
@@ -310,14 +310,69 @@ def abort(_: RaiseCancelT) -> Abort:
 
     @_public
     async def wait_readable(self, fd: int | _HasFileNo) -> None:
+        """
+        Block until the kernel reports that the given object is readable.
+
+        On Unix systems, ``obj`` must either be an integer file descriptor,
+        or else an object with a ``.fileno()`` method which returns an
+        integer file descriptor. Any kind of file descriptor can be passed,
+        though the exact semantics will depend on your kernel. For example,
+        this probably won't do anything useful for on-disk files.
+
+        On Windows systems, ``obj`` must either be an integer ``SOCKET``
+        handle, or else an object with a ``.fileno()`` method which returns
+        an integer ``SOCKET`` handle. File descriptors aren't supported,
+        and neither are handles that refer to anything besides a
+        ``SOCKET``.
+
+        :raises trio.BusyResourceError:
+            if another task is already waiting for the given socket to
+            become readable.
+        :raises trio.ClosedResourceError:
+            if another task calls :func:`notify_closing` while this
+            function is still working.
+        """
         await self._epoll_wait(fd, "read_task")
 
     @_public
     async def wait_writable(self, fd: int | _HasFileNo) -> None:
+        """Block until the kernel reports that the given object is writable.
+
+        See `wait_readable` for the definition of ``obj``.
+
+        :raises trio.BusyResourceError:
+            if another task is already waiting for the given socket to
+            become writable.
+        :raises trio.ClosedResourceError:
+            if another task calls :func:`notify_closing` while this
+            function is still working.
+        """
         await self._epoll_wait(fd, "write_task")
 
     @_public
     def notify_closing(self, fd: int | _HasFileNo) -> None:
+        """Call this before closing a file descriptor (on Unix) or socket (on
+        Windows). This will cause any `wait_readable` or `wait_writable`
+        calls on the given object to immediately wake up and raise
+        `~trio.ClosedResourceError`.
```

Comment:
First line should be a description but not exactly sure what would work.

Reply:
> Notify waiters of the given object that it will be closed.

works?

Reply:
Yeah